src/AdminBarApi.php

```diff
@@ -7,36 +7,96 @@
  */
 class AdminBarApi extends AbstractEndpoint {
 
-	/**
-	 * Endpoint path
-	 *
-	 * @Override
-	 * @var String
-	 */
-	protected $endpoint = '/admin-bar';
-
-	/**
-	 * Get the data.
-	 *
-	 * @Override
-	 * @param \WP_REST_Request $request The request.
-	 *
-	 * @return array|\WP_Error
-	 */
-	public function endpoint_callback( \WP_REST_Request $request ) {
-		global $wp_admin_bar;
-
-		$current_user = wp_get_current_user();
-
-		$data = [
-			'site_name' => get_bloginfo( 'name' ),
-			'user_name' => $current_user->user_login,
-			'dashboard_url' => admin_url(),
-			'logout_url' => wp_logout_url(),
-			'edit_page_url' => '',
-			'widgets' => $wp_admin_bar->get_node( 'new-content' ),
-		];
-
-		return $this->filter_data( $data );
-	}
+  /**
+   * Endpoint path
+   *
+   * @Override
+   * @var String
+   */
+  protected $endpoint = '/admin-bar';
+
+  /**
+   * Cookie name
+   *
+   * @var String
+   */
+  protected $cookieName = 'wp_admin_bar_user'; 
+
+  /**
+   * Cookie path
+   *
+   * @var String
+   */
+  protected $cookiePath = ''; 
+
+  /**
+   * Cookie domain
+   *
+   * @var String
+   */
+  protected $cookieDomain = '';
+
+  /**
+   * Constructor
+   */
+  function __construct() {
+    add_action('wp_login', [ $this, 'login_action' ], 10, 1);
+    add_action('wp_logout', [ $this, 'logout_action' ]);
+
+    $this->cookiePath = parse_url( get_option( 'siteurl' ), PHP_URL_PATH );
+    $this->cookieDomain = '.' . parse_url( get_option( 'siteurl' ), PHP_URL_HOST );
+  }
+
+  /**
+   * Login action listener
+   */
+  public function login_action( $user_login ) {
+    $adminBarUser = $_COOKIE[$this->cookieName];
+
+    if ( $adminBarUser != $user_login ) {
+      setcookie( $this->cookieName, $user_login, strtotime( '+1 day' ), $this->cookiePath, $this->cookieDomain );
+    }
+  }
+
+  /**
+   * Logout action listener
+   */
+  public function logout_action() {
+    setcookie( $this->cookieName, '', strtotime( '-1 day' ), $this->cookiePath, $this->cookieDomain );
+  }
+
+  /**
+   * Get the data.
+   *
+   * @Override
+   * @param \WP_REST_Request $request The request.
+   *
+   * @return array|\WP_Error
+   */
+  public function endpoint_callback( \WP_REST_Request $request ) {
+    $adminBarUser = $_COOKIE[$this->cookieName];
```

Comment:
@adamf321 

As I understand from the WP docs, we would need two requests to the endpoint for the get_currentuser() to work: the first one to generate a nonce, and the second to get the admin bar data using the nonce parameter. I think that the cookie method is better in this case, because we need it to know if the user is logged in or not at the frontend anyway. 

http://v2.wp-api.org/guide/authentication/#cookie-authentication
http://codex.wordpress.org/WordPress_Nonces